src/Ui/Component/Listing/Column/ProductSku.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace JustBetter\CustomerPricing\Ui\Component\Listing\Column;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Ui\Component\Listing\Columns\Column;
+
+class ProductSku extends Column
+{
+    protected $productRepository;
+
+    public function __construct(
+        ContextInterface $context,
+        UiComponentFactory $uiComponentFactory,
+        ProductRepositoryInterface $productRepository,
+        array $components = [],
+        array $data = []
+    ) {
+        $this->productRepository = $productRepository;
+        parent::__construct($context, $uiComponentFactory, $components, $data);
+    }
+
+    public function prepareDataSource(array $dataSource)
+    {
+        if (isset($dataSource['data']['items'])) {
+            foreach ($dataSource['data']['items'] as &$item) {
+                $productId = $item['product_id'];
+                $product = $this->productRepository->getById($productId);
```

Comment:
You could fetch the Sku by using the productResource directly instead of loading the complete product.

`$this->productResource->getProductsSku($productIds)`

You can view the use of this within the `Magento_InventoryCatalog` module.